ocaml/typing/includecore.ml

```diff
@@ -167,7 +98,9 @@ let value_descriptions ~loc env name
     loc
     vd1.val_attributes vd2.val_attributes
     name;
-  zero_alloc vd1.val_zero_alloc vd2.val_zero_alloc;
+  begin try Zero_alloc.sub_exn vd1.val_zero_alloc vd2.val_zero_alloc with
```

Comment:
I think this is fairly idiomatic for the compiler (e.g., this isn't the only try/with in this function). `sub_exn` is a series of checks any of which may fail independently, and I wanted to avoid writing a bunch of gross nested matches.  I could define a letop for bind in the result monad, or something, but in the typechecker the usual style is just to raise.

(I don't feel particularly strongly though - feel free to push back more)

Reply:
I thought all error creation appeared in tail position, except for one `if`-`then` needing to grow an `else`. If that’s true, let’s change it to return an error.

Reply:
Probably I'm missing some nice way to refactor this, but it's [these](https://github.com/ccasin/flambda-backend/blob/ccd41d34bd07feff3947fbaf2f96af0756789aac/ocaml/typing/zero_alloc.ml#L112) [two](https://github.com/ccasin/flambda-backend/blob/ccd41d34bd07feff3947fbaf2f96af0756789aac/ocaml/typing/zero_alloc.ml#L118) raises that seemed annoying.

Reply:
I missed the second one. Fine, then we should move the try-with into the module.

Reply:
Good idea, done.